compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -804,6 +804,24 @@ abstract class PrepJSInterop extends plugins.PluginComponent
             reporter.error(tree.pos,
                 "@JSBracketAccess methods with two parameters must return Unit")
           }
+
+          val flatParams = sym.paramss.flatten
+
+          for {
+            param <- flatParams
+            if isScalaRepeatedParamType(param.tpe)
+          } {
+            reporter.error(param.pos,
+                "@JSBracketAccess methods may not have repeated parameters")
+          }
+
+          for {
+            firstParam <- flatParams.headOption
+            if firstParam.isParamWithDefault
+          } {
+            reporter.error(firstParam.pos, "The first parameter of a " +
+                "@JSBracketAccess method may not be a default parameter")
```

Comment:
It will toggle between assignment and select. It is our own little support for jQuery style APIs :P

I'm fine with forbidding it (can always use an overload, which is probably better anyways). This is just one piece that worked before.